### PR TITLE
[release/6.0] Backport Fix issues with hosting startups

### PR DESCRIPTION
**Pull request by davidfowl**
Backport of https://github.com/dotnet/aspnetcore/pull/36055

## Customer Impact
Hosting startups in the libraries that don't match the entry point don't run. This affects tests but it also affects libraries.

## Testing

Unit tests.

## Risk

Low

### CODE REVIEW

**Comment by davidfowl** [comment]
cc @dougbu @wtgodbe

**Comment by davidfowl** [comment]
> What do you need here @davidfowl❔

To hit that merge button.

**Review by BrennanConroy — APPROVED** [review]

**Comment by dougbu** [comment]
> To hit that merge button.

Please ping us after approvals (fine here on that) and the build is green (not). Thanks

**Comment by davidfowl** [comment]
> Please ping us after approvals (fine here on that) and 

It's approved.

> the build is green (not).

The build is flaky.

**Comment by davidfowl** [comment]
@dougbu can you mark this as auto merge then? That way I can re-run the build and not ping you again once it passes.

**Comment by dougbu** [comment]
Done. Sorry, I didn't look into the build failure.

**Pull request merged**
